### PR TITLE
feat: add `constants/float32/ln-half`

### LINKED ISSUE #3332: [RFC]: add `constants/float32/ln-half`
### Description

This RFC proposes
- This RFC proposes adding the package @stdlib/constants/float32/ln-half returning Natural Logarithm of single-precision floating-point number 1/2.

Image

### Related Issues

Related issues # , # , and # .

Related issues None.

### Questions

No.

### Other

No.

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by vivekmaurya001**
Resolves #3332 .

## Description

> What is the purpose of this pull request?

This pull request:

-   This RFC proposes adding the package @stdlib/constants/float32/ln-half returning Natural Logarithm of single-precision floating-point number 1/2.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #{{TODO: add issue number}} 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/constants/float32/ln-half/README.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# FLOAT32_LN_HALF
+
+> [Natural logarithm][@stdlib/math/base/special/ln] of single-precision floating-point number `1/2`.
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/constants/float32/ln-half/README.md]: @vivekmaurya001 Can you please update this description here and everywhere else in this package to:

```
Natural logarithm of `1/2` as a single-precision floating-point number.
```

File: lib/node_modules/@stdlib/constants/float32/ln-half/README.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+console.log( FLOAT32_LN_HALF );
+// => -0.6931471824645996
+```
+
+</section>
+
+<!-- /.examples -->
+
+<!-- C 6931472 documentation. -->
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/constants/float32/ln-half/README.md]:
```suggestion
<!-- C interface documentation. -->
```

File: lib/node_modules/@stdlib/constants/float32/ln-half/README.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+## Usage
+
+```javascript
+var FLOAT32_LN_HALF = require( '@stdlib/constants/float32/ln-half' );
+```
+
+#### FLOAT32_LN_HALF
+
+[Natural logarithm][@stdlib/math/base/special/ln] of `1/2`.
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/constants/float32/ln-half/README.md]: @vivekmaurya001 Also, you'll need to update this path to the single-precision equivalent, i.e., `lnf` here and everywhere else in this file.

File: lib/node_modules/@stdlib/constants/float32/ln-half/README.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+#### STDLIB_CONSTANT_FLOAT32_LN_HALF
+
+Macro for the [natural logarithm][@stdlib/math/base/special/ln] of single-precision floating-point number `1/2`.
+
+</section>
+
+<!-- /.usage -->
+
+<!-- C 6931472 usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/constants/float32/ln-half/README.md]:
```suggestion
<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
```

File: lib/node_modules/@stdlib/constants/float32/ln-half/README.md (hunk 1 above)
Comment (vivekmaurya001) [reply lib/node_modules/@stdlib/constants/float32/ln-half/README.md]: done the changes !

File: lib/node_modules/@stdlib/constants/float32/ln-half/README.md (hunk 3 above)
Comment (vivekmaurya001) [reply lib/node_modules/@stdlib/constants/float32/ln-half/README.md]: done it !

File: lib/node_modules/@stdlib/constants/float32/ln-half/test/test.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+tape( 'main export is a number', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof FLOAT32_LN_HALF, 'number', 'main export is a number' );
+	t.end();
+});
+
+tape( 'export is a single-precision floating-point number equal to `-0.6931471824645996', function test( t ) {
+	var expected = lnf( 0.5 );
+	t.equal( FLOAT32_LN_HALF, expected, 'returns expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/constants/float32/ln-half/test/test.js]:
```suggestion
	t.equal( FLOAT32_LN_HALF, -0.6931471824645996, 'returns expected value' );
```

The intent here is to test the value exactly. The computed value may not be exact and depends on the upstream `lnf`, which is not what we want for this test case.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @vivekmaurya001.

**Pull request merged**